### PR TITLE
some refactoring and fixes some violations.

### PR DESCRIPTION
**Pull request by mkouhei**
- Does refactoring.
  - Supports Python 3.5
  - Pythonic code sytle.
- Fixes some violations.
  - pylint
  - pep8
  - pep257

Merge this pull request if it is fine with you.
Best regards, :)

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -8,9 +8,8 @@ Let's eliminate `cd` and `source` in your crontab!
 Requirements
 ============
 
-* Supports Python 2.7 only
-
-We'll be supporting 3.5 soon.
+* Supports Python 2.7, 3.5.
+* Bash.
```
Comment (sugitak) [inline README.rst:12]: :)

File: appdo/appdo.py [hunk 2]
```diff
@@ -14,194 +13,160 @@
...
-        try:
-            cmd = conf['cd']
-            if isinstance(cmd, str):
-                return ['cd ' + cmd]
-            else:
-                return []
-        except KeyError:
+    def _build_cd_command(conf):
+        """private method."""
+        cmd = conf['cd']
```
Comment (sugitak) [inline appdo/appdo.py]: so this should be `conf.get('cd')`, right?

File: appdo/appdo.py [hunk 3]
```diff
@@ -14,194 +13,160 @@
...
-            cmd = conf['before']
-            if isinstance(cmd, str):
-                return [cmd]
-            else:
-                return cmd
-        except KeyError:
-            return []
+    def _build_before_commands(conf):
+        """private method."""
+        cmd = conf['before']
```
Comment (sugitak) [inline appdo/appdo.py]: and this `conf.get('before')`

File: appdo/appdo.py [hunk 4]
```diff
@@ -14,194 +13,160 @@
...
-        try:
-            envs = conf['env']
-            if isinstance(envs, dict):
-                return [k + '=' + v for k,v in envs.items()]
-            else:
-                return []
-        except KeyError:
+    def _build_envs(conf):
+        """private method."""
+        envs = conf['env']
```
Comment (sugitak) [inline appdo/appdo.py]: and this `conf.get('env')`

File: appdo/appdo.py [hunk 5]
```diff
@@ -14,194 +13,160 @@
...
-        try:
-            cmd = conf['prefix']
-            if isinstance(cmd, str):
-                return [cmd + ' ']
-            else:
-                return []
-        except KeyError:
+    def _build_prefix_command(conf):
+        """private method."""
+        cmd = conf['prefix']
```
Comment (sugitak) [inline appdo/appdo.py]: and `conf.get('prefix')`.

I should have written this in higher-order function manner...

**Comment by sugitak** [comment]
Thank you so much for your great fixes, they're really great!
I've checked out each of them. Hope I could be a real Pythonista some day.

Well, first, I'd be adding tests... and then split the files...

File: appdo/appdo.py (hunk 2 above)
Comment (mkouhei) [inline appdo/appdo.py]: Yes, I'm sorry :)

File: appdo/appdo.py (hunk 3 above)
Comment (mkouhei) [inline appdo/appdo.py]: Yes.

**Comment by mkouhei** [comment]
@sugitak Fixes `.get('key')` instead of `['key']`.

**Comment by sugitak** [comment]
Thank you so much for your awesome help!

**Pull request merged**